lib/dal/src/audit_logging.rs

```diff
@@ -184,50 +198,93 @@ pub async fn list(ctx: &DalContext) -> Result<Vec<FrontendAuditLog>> {
+    let mut messages = consumer.fetch().max_messages(1000).messages().await?;
     let mut frontend_audit_logs = Vec::new();
 
     while let Some(Ok(message)) = messages.next().await {
         let audit_log: AuditLog = serde_json::from_slice(&message.payload)?;
-        match audit_log {
-            AuditLog::V2(inner) => {
-                // TODO(nick): cache users.
-                let (user_id, user_email, user_name) = match inner.actor {
-                    Actor::System => (None, None, None),
-                    Actor::User(user_id) => {
-                        let user = User::get_by_pk(ctx, user_id.into())
-                            .await
-                            .map_err(Box::new)?
-                            .ok_or(AuditLoggingError::UserNotFound(user_id.into()))?;
-                        (
-                            Some(user_id),
-                            Some(user.email().to_owned()),
-                            Some(user.name().to_owned()),
-                        )
-                    }
-                };
-                frontend_audit_logs.push(FrontendAuditLog {
-                    user_id,
-                    user_email,
-                    user_name,
-                    kind: inner.kind.to_string(),
-                    timestamp: inner.timestamp,
-                    change_set_id: inner.change_set_id,
-                    metadata: serde_json::to_value(FrontendAuditLogDeserializedMetadata::from(
-                        inner.kind,
-                    ))?,
-                });
-            }
-            AuditLog::V1(_) => {
-                debug!("skipping older audit logs in beta...");
-            }
+        if let Some(frontend_audit_log) = assemble_for_list(ctx, audit_log).await? {
+            frontend_audit_logs.push(frontend_audit_log);
         }
     }
 
     Ok(frontend_audit_logs)
 }
 
+async fn assemble_for_list(
+    ctx: &DalContext,
+    audit_log: AuditLog,
+) -> Result<Option<FrontendAuditLog>> {
+    match audit_log {
+        AuditLog::V3(inner) => {
+            // TODO(nick): cache change sets.
+            let (change_set_id, change_set_name) = match inner.change_set_id {
+                Some(change_set_id) => {
+                    let change_set = ChangeSet::find(ctx, change_set_id.into())
+                        .await
+                        .map_err(Box::new)?
+                        .ok_or(AuditLoggingError::ChangeSetNotFound(change_set_id.into()))?;
+                    match change_set.status {
+                        ChangeSetStatus::Abandoned
+                        | ChangeSetStatus::Failed
+                        | ChangeSetStatus::Rejected => {
+                            trace!(?change_set.status, ?change_set.id, "skipping change set for audit log assembly due to status");
+                            return Ok(None);
+                        }
+                        ChangeSetStatus::Applied
```

Comment:
Right now, viewing events for a `Applied` change sets might be weird on HEAD since the filters are broken (known) at the moment. If you clear the filters, you'll be able to see the events, but it will be unclear that they are on HEAD.